docs/code-style.md

```diff
@@ -0,0 +1,333 @@
+# Code Style
+
+---
+
+**NOTE**:
+
+Most of the information in this document is meant to help keep the code looking
+similar and predictable while being worked on by different people. It isn't
+meant to be rigid and static, but a suggestion. Thus, rule of thumb:
+
+1. use your own best judgement;
+2. try to follow style of code already in place;
+3. notify others of issues and possible improvements.
+
+---
+
+## Comments (general)
+
+There usually isn't a formal way of writing comments. Most common are simple
+line comments, not block ones. However, few tags, due to familiarity with JSDoc,
+can be found outside these `/** */` blocks:
+
+- `@debt` is usually some future issue or pull request to repay with better code
+  solution be it refactor or replacement;
+- `@deprecated` is usually a comment that gives an explanation and alternative;
+- `@see` is usually followed by URL or similar reference for further info.
+
+## Style Sheets (coding/design)
+
+Over the course of the codebase there has been a revolving door of different
+(S)CSS and CSS-in-JS systems. This is an attempt to give a general overview of
+how they (can) coexist.
+
+### Previous Different systems
+
+There has been a mix of different style systems in use in the past. Some
+include:
+
+- _Styled Components_
+- SCSS provided by _Create React App_ (_CRA_), not including modules
+- _Bootstrap_ styles
+- global overrides for the most basic HTML tags
+
+### Naming
+
+The naming of classes followed BEM (more at
+https://en.bem.info/methodology/quick-start).
+
+With a minor adjustment of using the `PascalCase` for the top level B(lock) to
+agree with the name of the component.
+
+### SCSS specific functionalities
+
+Most of them could be found at `./src/scss`. More specifically the `colors.scss`
+and `constants.scss` files.
+
+Use of `@import` is deprecated by SCSS standard while encouraging replacement
+with `@use`. The latter provides for better encapsulation and namespacing.
+
+### SCSS modules + Tailwind parity
+
+In order to make code style more uniform while using two different systems, here
+are some naming conventions that can be used:
+
+- import `classnames` library as `cn` for brevity since it will be used
+  extensively;
+- import Tailwind class strings from another file with `import * as TW from`;
+- import SCSS module class maps as `import CN from`;
```

Comment:
I think `cn` and `CN` is going to lead to confusion in the future.

Reply:
Maybe.  Anything that's close to `TW` as a short and all caps due to being a constant would work. 
`M` for module? `CSS`? 

I'd argue low caps and all caps is indicative enough of what is function and what is constant even before compiler does the check. Still, the document is written in a way to not be mandating anything, but more like giving an example.

So, if there is a possibility of confusion, we can add a line about that, maybe provide different naming ideas in it.

I will merge this as is now and we can update it with the next idea we get after the experience.